app/src/main/java/com/example/recipeapp/models/Recipe.java

```diff
@@ -103,4 +140,124 @@ public void setReviews(List<Review> reviews) {
+            }
+            recipe.setInstructions(instructions);
+
+            //TODO: Add instructions, image file, and author
+//            String imageURL = results.getJSONObject(i).getString("image");
+//            Log.i(TAG, imageURL);
+//            File resize = new File("photo.jpg");
+//            try {
+//                java.net.URL img_value = new java.net.URL(imageURL);
+//                Bitmap mIcon = BitmapFactory.decodeStream(img_value.openConnection()
+//                                .getInputStream());
+//                if (mIcon != null) {
+//                    byte[] imgByteArray = encodeToByteArray(mIcon);
+//                    resize = recipe.resizeFile(mIcon);
+//                }
+//                Log.i(TAG,"Successfully saved file");
+//            } catch (Exception e) {
+//                e.printStackTrace();
+//            }
+//            recipe.setImage(new ParseFile(resize));
+
+            recipe.setImageUrl(results.getJSONObject(i).getString("image"));
+//            if(!recipe.isRecipeStored()) {
+//                recipe.saveInBackground(new SaveCallback() {
+//                    @Override
+//                    public void done(ParseException e) {
+//                        if (e != null) {
+//                            Log.e(TAG, "Unable to save recipe! ", e);
+//                        } else
+//                            Log.i(TAG, "Added " + recipe.getTitle() + " to database!");
+//                    }
+//                });
+//
+//            }
+            Log.i(TAG, "Added " + recipe.getTitle());
+            recipes.add(recipe);
+
+        }
+        return recipes;
+    }
+
+    public static Recipe addIngredients(Recipe recipe,List<String> ingredients) throws JSONException {
+        recipe.setIngredientList(ingredients);
+        Log.i(TAG,"Rec: " + recipe.getIngredientList().toString());
+        return recipe;
+    }
+
+    public File resizeFile(Bitmap image) {
+        Bitmap resizedBitmap = BitmapScaler.scaleToFitWidth(image, 800);
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        resizedBitmap.compress(Bitmap.CompressFormat.JPEG, 40, bytes);
+        File resizedFile = new File("photo.jpg");
+        try {
+            resizedFile.createNewFile();
+        } catch (IOException e) {
+            Log.e(TAG, "Unable to create new file ", e);
+        }
+        FileOutputStream fos = null;
+        try {
+            fos = new FileOutputStream(resizedFile);
+        } catch (FileNotFoundException e) {
+            Log.e(TAG, "File not found ", e);
+        }
+        try {
+            fos.write(bytes.toByteArray());
+        } catch (IOException e) {
+            Log.e(TAG, "Unable to write to file ", e);
+        }
+        try {
+            fos.close();
+        } catch (IOException e) {
+            Log.e(TAG, "Unable to close file ", e);
+        }
+        Log.i(TAG, "File: " + resizedFile);
+        return resizedFile;
+    }
```

Comment:
Maybe consider combining try / catch blocks since each ostensibly is depended on by the one preceding it. Plus - if any of these go wrong, wouldn't we want to just return null altogether? just something to consider.